src/lightning/fabric/strategies/fsdp.py

```diff
@@ -364,6 +371,12 @@ def save_checkpoint(
                 "`FSDPStrategy.save_checkpoint(..., storage_options=...)` is not supported because"
                 " `FSDPStrategy` does not use the `CheckpointIO`."
             )
+        if filter is not None and self._state_dict_type == "sharded":
+            # A KeyError is raised when `torch.distributed.checkpoint.load_state_dict` is called
+            raise NotImplementedError(
+                "FSDP doesn't support loading sharded filtered checkpoints, so saving them is disabled."
+            )
```

Comment:
Could you open an issue about this?

Reply:
Do you mean in PyTorch?

Reply:
Since you added this explicit error, I assume we are missing the proper handling? We could open an issue on our side to investigate.

Reply:
Opened https://github.com/pytorch/pytorch/issues/105379